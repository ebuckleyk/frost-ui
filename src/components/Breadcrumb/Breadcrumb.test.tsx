import * as React from 'react';
import { render } from '@testing-library/react';

import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbList,
  BreadcrumbPage,
  BreadcrumbSeparator,
} from './Breadcrumb';

const Component = () => {
  return (
    <Breadcrumb>
      <BreadcrumbList>
        <BreadcrumbItem>
          <BreadcrumbLink href="/">Home</BreadcrumbLink>
        </BreadcrumbItem>
        <BreadcrumbSeparator />
        <BreadcrumbItem>
          <BreadcrumbLink href="/components">Components</BreadcrumbLink>
        </BreadcrumbItem>
        <BreadcrumbSeparator />
        <BreadcrumbItem>
          <BreadcrumbPage>Breadcrumb</BreadcrumbPage>
        </BreadcrumbItem>
      </BreadcrumbList>
    </Breadcrumb>
  );
};

describe('Breadcrumb', () => {
  it('should render Breadcrumb component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
