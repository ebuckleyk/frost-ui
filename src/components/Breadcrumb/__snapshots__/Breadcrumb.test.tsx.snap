// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb > should render Breadcrumb component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="flex flex-wrap items-center gap-1.5 break-words text-sm text-muted-foreground sm:gap-2.5"
        >
          <li
            class="inline-flex items-center gap-1.5"
          >
            <a
              class="transition-colors hover:text-foreground"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="[&>svg]:size-3.5"
            role="presentation"
          >
            <svg
              class="lucide lucide-chevron-right"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </li>
          <li
            class="inline-flex items-center gap-1.5"
          >
            <a
              class="transition-colors hover:text-foreground"
              href="/components"
            >
              Components
            </a>
          </li>
          <li
            aria-hidden="true"
            class="[&>svg]:size-3.5"
            role="presentation"
          >
            <svg
              class="lucide lucide-chevron-right"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </li>
          <li
            class="inline-flex items-center gap-1.5"
          >
            <span
              aria-current="page"
              aria-disabled="true"
              class="font-normal text-foreground"
              role="link"
            >
              Breadcrumb
            </span>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="flex flex-wrap items-center gap-1.5 break-words text-sm text-muted-foreground sm:gap-2.5"
      >
        <li
          class="inline-flex items-center gap-1.5"
        >
          <a
            class="transition-colors hover:text-foreground"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-hidden="true"
          class="[&>svg]:size-3.5"
          role="presentation"
        >
          <svg
            class="lucide lucide-chevron-right"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </li>
        <li
          class="inline-flex items-center gap-1.5"
        >
          <a
            class="transition-colors hover:text-foreground"
            href="/components"
          >
            Components
          </a>
        </li>
        <li
          aria-hidden="true"
          class="[&>svg]:size-3.5"
          role="presentation"
        >
          <svg
            class="lucide lucide-chevron-right"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
        </li>
        <li
          class="inline-flex items-center gap-1.5"
        >
          <span
            aria-current="page"
            aria-disabled="true"
            class="font-normal text-foreground"
            role="link"
          >
            Breadcrumb
          </span>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
